types/src/validator_verifier.rs

```diff
@@ -344,51 +351,63 @@ impl ValidatorVerifier {
     }
 
     /// Returns quorum voting power.
-    pub fn quorum_voting_power(&self) -> u64 {
+    pub fn quorum_voting_power(&self) -> u128 {
         self.quorum_voting_power
     }
 
     /// Returns total voting power.
-    pub fn total_voting_power(&self) -> u64 {
+    pub fn total_voting_power(&self) -> u128 {
         self.total_voting_power
     }
 }
 
 /// Returns sum of voting power from Map of validator account addresses, validator consensus info
-fn sum_voting_power(
-    address_to_validator_info: &BTreeMap<AccountAddress, ValidatorConsensusInfo>,
-) -> u64 {
-    address_to_validator_info.values().fold(0, |sum, x| {
-        sum.checked_add(x.voting_power)
+fn sum_voting_power(address_to_validator_info: &[ValidatorConsensusInfo]) -> u128 {
+    address_to_validator_info.iter().fold(0, |sum, x| {
+        sum.checked_add(x.voting_power as u128)
             .expect("sum of all voting power is greater than u64::max")
     })
 }
 
 impl fmt::Display for ValidatorVerifier {
     fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
         write!(f, "ValidatorSet: [")?;
-        for (addr, info) in &self.address_to_validator_info {
-            write!(f, "{}: {}, ", addr.short_str_lossless(), info.voting_power)?;
+        for info in &self.validator_infos {
+            write!(
+                f,
+                "{}: {}, ",
+                info.address.short_str_lossless(),
+                info.voting_power
+            )?;
         }
         write!(f, "]")
     }
 }
 
+/// This does the conversion between move data to the rust data
 impl From<&ValidatorSet> for ValidatorVerifier {
     fn from(validator_set: &ValidatorSet) -> Self {
-        ValidatorVerifier::new(validator_set.payload().fold(
-            BTreeMap::new(),
-            |mut map, validator| {
-                map.insert(
-                    *validator.account_address(),
+        let sorted_validator_infos: BTreeMap<u64, ValidatorConsensusInfo> = validator_set
```

Comment:
Can you add an UT for this if there is not already one?

Reply:
there're many tests does this conversion, and we have assertion in place to make sure it retains correct index